### ISSUE TITLE
Span generated when adding Dynamic Data resulting in HTML error

### ISSUE THREAD

**Issue by mamaddi** [soflyy/oxygen-bugs-and-features#315]
**Describe the bug**
When adding Dynamic Data (for instance Content) te builder creates a **div** and a **span** element. So when the content has a block element (such as p) the result is an HTML error (a p element nested inside a span element).
<img width="784" alt="Schermata 2019-05-20 alle 10 37 50" src="https://user-images.githubusercontent.com/47536545/58007839-6613fd00-7aeb-11e9-9135-446a74ef1e5f.png">

**Have you tried all the steps at https://oxygenbuilder.com/documentation/troubleshooting/troubleshooting-guide/?**
*Yes*

**Are you able to replicate the issue on a Sandbox install at https://oxygenbuilder.com/try?**
Yes: http://gleaming-fish.w6.wpsandbox.pro/2017/03/09/hello-world/

**To Reproduce**
Add Dynamic Data and look at the code

**Expected behavior**
No span element should be generated, to avoid HTML errors.

**What actually happens**
The builder outputs a span element.

**Desktop (please complete the following information):**
- OS: MacOS
 - Browser: Chrome
 - Oxygen Version: 2.3

**Comment by LordWhisper** [comment]
Thanks Maddalena for raising the bug here. 

Been mention it in the group before but didn't have the time to replicate it to create the bug report: https://www.facebook.com/groups/1626639680763454/permalink/2085851074842310/

**Comment by KittenCodes** [comment]
Replicated in 2.4.1.

**Comment by adrien-robert** [comment]
+1 for fixing this in the next version please.
As it's already a "confirmed bug".

Thanks Oxygen team! =)

**Comment by bpages383** [comment]
Got this one too on my latest project. Hopping for a fix soon :) thanks for your hard work Oxygen team.

**Comment by Grandiferr** [comment]
Just tried working with Oxygen. This is the single most annoying thing that bothers me. The issue is open since may 2019. How is it not fixed yet?

**Comment by jamescrabb** [comment]
Needs addressing. Not producing valid html5 is not good

**Comment by Grandiferr** [comment]
3.6 alpha is available and the bug is still there.

**Comment by richardroscoe** [comment]
+1

**Comment by arsh999cg** [comment]
+1

**Comment by wpsumo** [comment]
+1

**Comment by LauGau** [comment]
+999

**Comment by Grandiferr** [comment]
Nice job, guys! Keep wasting money on ads and not solving the problems!

**Comment by brend2001** [comment]
is there any alternative solution to this problem guys?

**Comment by Pab89** [comment]
+1

**Comment by taenzerme** [comment]
When will this be fixed?

**Comment by chris2604** [comment]
+1 Still not fixed since 2 years

**Comment by chris2604** [comment]
In a project I used Dynamic Data > Content to integer the content of our pages into a template. A span was wrapping all the content page.
To fix this, instead of the Dynamic Data > Content, I used a Code block with the following content : 
<?php the_content(); ?>

In our case, the result was the same, but without the span.

Hope it will help !

**Comment by schietegal** [comment]
+1

**Comment by wpsumo** [comment]
+1

**Comment by cerontek** [comment]
+1

**Comment by ikx94** [comment]
Please fix it.

**Comment by maltmann-muc** [comment]
+1

**Comment by antic1pate** [comment]
+1

**Comment by perrelet** [comment]
+1

**Comment by Mozzarellathicc** [comment]
+1

**Comment by Yosi11** [comment]
This is still happening in Version 3.9, please fix this.

**Comment by AztecInternet** [comment]
This kinda sucks and RankMath can't seem to analyse my site properly either because of this terrible HTML error. 

Come on guys. Fix it already!!! This is terrible.

**Comment by LauGau** [comment]
Yes, having a <p> inside a <span> is pretty ugly...

**Comment by chris3016** [comment]
Brutal...

**Comment by wpsumo** [comment]
Yeah same here use code block for everything as the native elements adding IDs non-necessary IDs, Divs and Spans. Haven't used many native functions in oxygen for a long time as it's messy with all this. It's get enough of DOM elements in a design and code block helps us reduce it and keep it clean. Thanks for code block, but it would make sense to fix the native functions and not force span, divs and IDs.

**Comment by IKuipers1** [comment]
I also asked this to Oxygen like a week ago. This was their answer. 

I'm very sorry for the inconvenience. Our development is aware of the issue, but due to the complexity of the issue, there's currently no fix for it. 

This is an issue that our development team is aware of and investigating. Unfortunately, I cannot provide further details as to when it may be addressed.

**Comment by greggdev** [comment]
Still happening on 4.0, make is difficult or impossible to work with dynamic data. And what is that response, "complexity of the issue"....how come other site builders don't have this issue? and not fixed for how many years now? This post originated May 20, 2019.

**Comment by kvler** [comment]
Can someone explain the code block work around for this issue? I want to insert data from a custom field (ACF) and not have it wrapped in a span. Thanks.

**Comment by MikeJoe2602** [comment]
+1

**Comment by gerogr** [comment]
+1

**Comment by lucianahanan** [comment]
+1

**Comment by dmduco** [comment]
+1

**Comment by MaximeStarssed** [comment]
+1

**Comment by dywoox** [comment]
This “problem” mainly affects the input of dynamic content. 
the <span> tag can be bypassed if the variables such as [oxygen data='title'] [oxygen data='post_type'] ect. are entered in the text field. however, you will then only see the variable in the backend and not the content.

Another possibility is to use the shortcode element and enter the variables. The only thing that bothers me is that the ID and the automatic class contain the term shortcode. another ID and class addition would be more suitable for texts. 

Perhaps a shortcode clone would be a simple solution: which is only intended for dynamic content or texts. If you don't need the <span> for styling and know what you are doing, you could use this function.

The other advantage would also be that the automatically created spans are not “touched” in existing installations.
Alternatively, a function that is switched off by default would be whether automatic spans should be created for newly created fields. 

The question I ask myself: what would be the easiest and safest for existing installations with the least effort? I think a new extension that can create dynamic fields, without <span> like shortcode, but which can also display inscribed text in the backend (which shortcode cannot do).
